lib/custom_markdown_renderer.rb

```diff
@@ -0,0 +1,97 @@
+class CustomMarkdownRenderer < CommonMarker::HtmlRenderer
+  YOUTUBE_REGEX = %r{https?://(?:www\.)?(?:youtube\.com/watch\?v=|youtu\.be/)([^&/]+)}
+  VIMEO_REGEX = %r{https?://(?:www\.)?vimeo\.com/(\d+)}
+  MP4_REGEX = %r{https?://(?:www\.)?.+\.(mp4)}
+
+  def text(node)
+    content = node.string_content
+
+    # Check for presence of '^' in the content
+    if content.include?('^')
+      # Split the text and insert <sup> tags where necessary
+      split_content = parse_sup(content)
+      # Output the transformed content
+      out(split_content.join)
+    else
+      # Output the original content
+      out(escape_html(content))
+    end
+  end
+
+  def link(node) # rubocop:disable Metrics/MethodLength
+    link_url = node.url
+
+    youtube_match = link_url.match(YOUTUBE_REGEX)
+    if youtube_match
+      embed_code = make_youtube_embed(youtube_match)
+      out(embed_code)
+      return
+    end
+
+    vimeo_match = link_url.match(VIMEO_REGEX)
+    if vimeo_match
+      embed_code = make_vimeo_embed(vimeo_match)
+      out(embed_code)
+      return
+    end
+
+    mp4_match = link_url.match(MP4_REGEX)
+    if mp4_match
+      embed_code = make_video_embed(link_url)
+      out(embed_code)
+      return
+    end
+
+    # If it's not YouTube or Vimeo link, render normally
+    super
+  end
```

Comment:
rather than disabling the rubocop here, we can break the method into sub methods

Reply:
@tejaswinichile the return in those cases is needed; I have pushed the refactored changes. Please check cc: @sojan-official